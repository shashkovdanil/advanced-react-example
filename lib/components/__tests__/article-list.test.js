import React from 'react';
import ArticleList from '../article-list';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {
        id: 'a',
        title: 'a',
        date: 'a',
        authorId: 'a',
        body: 'a'
      },
      b: {
        id: 'b',
        title: 'b',
        date: 'b',
        authorId: 'b',
        body: 'b'
      }
    }
  };
  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );
    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
