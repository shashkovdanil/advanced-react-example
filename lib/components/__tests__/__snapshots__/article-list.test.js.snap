// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "authorId": "a",
          "body": "a",
          "date": "a",
          "id": "a",
          "title": "a",
        },
        "b": Object {
          "authorId": "b",
          "body": "b",
          "date": "b",
          "id": "b",
          "title": "b",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArticleContainer
          article={
            Object {
              "authorId": "a",
              "body": "a",
              "date": "a",
              "id": "a",
              "title": "a",
            }
          }
        />,
        <ArticleContainer
          article={
            Object {
              "authorId": "b",
              "body": "b",
              "date": "b",
              "id": "b",
              "title": "b",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "authorId": "a",
            "body": "a",
            "date": "a",
            "id": "a",
            "title": "a",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "authorId": "b",
            "body": "b",
            "date": "b",
            "id": "b",
            "title": "b",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArticleContainer
            article={
              Object {
                "authorId": "a",
                "body": "a",
                "date": "a",
                "id": "a",
                "title": "a",
              }
            }
          />,
          <ArticleContainer
            article={
              Object {
                "authorId": "b",
                "body": "b",
                "date": "b",
                "id": "b",
                "title": "b",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "authorId": "a",
              "body": "a",
              "date": "a",
              "id": "a",
              "title": "a",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "authorId": "b",
              "body": "b",
              "date": "b",
              "id": "b",
              "title": "b",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
